package-res/reportviewer/paginationTemplate.html

```diff
@@ -238,6 +240,21 @@
     );
   }
 
+  function toLocaleStringSupportsLocales() {
+  var number = 0;
+  try {
+    number.toLocaleString('i');
+  } catch (e) {
+    return e.name === 'RangeError';
```

Comment:
This method here does not actually handle the case when the method "toLocaleString" does not exist. 

When the method itself is not supported, you wont get a "range-error", you get a different error depending on what browser you use. So checking for a specific error code is rather optimistic - especially when we are dealing with really really old browsers or email clients.

Play it safe and catch everything. 

And why test for the options-argument when your real method call never uses it? All you need here is to check whether the call to the method with no parameters will succceed. If you dont need the extended options, dont test for them.


As for the different behaviour between PUC and this template:

Locale handling in PUC is implemented via Dojo. As this is a standalone HTML file, with no server to provide libraries, we have to rely on whatever is built into the browser. The "toLocaleString" method is supported in all the latest browsers, but older browsers and email clients may not have that method. 

If a report is scheduled and has enough rows to trigger the auto-scheduling, the exporter will use this template to produce the report - so we better make sure that we are as compatible as possible. 

And yes, our behaviour will not be strictly the same as in the BI server if the user happens to use an old HTML viewer. But with the combination of "toLocaleString" method as default and the fallback to the hard-coded "en-US" style string format, we at least try our best and return a sensible fall-back behaviour. 



And as I said in the other comment: once you have formatted the number in the try block - why throw the result away? Use it.